File: Samples/JotFormTest/Classes/Controller/CreateSubmissionViewController.m
//
//  CreateSubmissionViewController.m
//  JotFormTest
//
//  Created by Interlogy, LLC on 1/7/14.
//  Copyright (c) 2014 Interlogy, LLC. All rights reserved.
//

#import "CreateSubmissionViewController.h"
#import "SVProgressHUD.h"
#import "SharedData.h"
#import "Common.h"
#import <JotForm_iOS/JotForm.h>

@interface CreateSubmissionViewController ()

@end

@implementation CreateSubmissionViewController

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    
    self.title = @"Create submission";
}


#pragma mark - User definition method

- (void)createSubmission {
    [SVProgressHUD showWithStatus:@"Creating submissions..."];
    
    NSMutableDictionary *submission = [[NSMutableDictionary alloc] init];
    submission[@"1"] = @"XXX";
    submission[@"2"] = @"This is a test for creating submission.";
    
    [[SharedData sharedData].apiClient createFormSubmissions:FORM_ID submission:submission onSuccess:^(id result) {
        [SVProgressHUD dismiss];
        
        if (result) {
            NSInteger responseCode = [result[@"responseCode"] integerValue];
            
            if (responseCode == 200 || responseCode == 206) {
                UIAlertController *alertView = [UIAlertController
                                                alertControllerWithTitle:@"JotFormAPISample"
                                                message:@"You created submission successfully"
                                                preferredStyle:UIAlertControllerStyleAlert];
                
                UIAlertAction *cancelButton = [UIAlertAction
                                               actionWithTitle:@"Ok"
                                               style:UIAlertActionStyleDefault
                                               handler:^(UIAlertAction *action){
                                                   
                                               }];
                
                [alertView addAction:cancelButton];
                [self presentViewController:alertView animated:YES completion:nil];
                
                [self.navigationController popViewControllerAnimated:YES];
            }
        }
      } onFailure:^(id error) {
   
      }];
}

#pragma mark - IBAction

- (IBAction)createSubmissionButtonClicked:(id)sender {
    [self createSubmission];
}

@end

File: Samples/JotFormTest/Classes/Controller/CreateSubmissionViewController.swift
//
//  CreateSubmissionViewController.swift
//  JotFormTest
//
//  Created by Curtis Stilwell on 2/4/18.
//  Copyright © 2018 wang. All rights reserved.
//

import Foundation

class CreateSubmissionViewController: UIViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view from its nib.
        title = "Create submission"
    }
    
    func createSubmission() {
        SVProgressHUD.show(withStatus: "Creating submissions...")
        var submission = [String:Any]()
        submission["1"] = "XXX"
        submission["2"] = "This is a test for creating submission."
        
        SharedData.sharedData.apiClient?.createFormSubmissions(FORM_ID, submission: submission, onSuccess: {(_ result: Any) -> Void in
            SVProgressHUD.dismiss()
            
            let alertView = UIAlertController(title: "JotFormAPISample", message: "You created submission successfully", preferredStyle: .alert)
            let cancelButton = UIAlertAction(title: "Ok", style: .default, handler: {(_ action: UIAlertAction) -> Void in
            })
            
            alertView.addAction(cancelButton)
            self.present(alertView, animated: true, completion: nil)
            
        }, onFailure: {(_ error: Any) -> Void in
            SVProgressHUD.dismiss()
        })
    }
    
    @IBAction func createSubmissionButtonClicked(_ sender: Any) {
        createSubmission()
    }
    
}
